Source/Request.swift

```diff
@@ -414,3 +323,283 @@ extension Request: CustomDebugStringConvertible {
         return components.joined(separator: " \\\n\t")
     }
 }
+
+// MARK: -
+
+/// Specific type of `Request` that manages an underlying `URLSessionDataTask`.
+open class DataRequest: Request {
+
+    // MARK: Helper Types
+
+    enum Requestable: TaskConvertible {
+        case request(URLRequest)
+
+        func task(session: URLSession, adapter: RequestAdapter?, queue: DispatchQueue) -> URLSessionTask {
+            var task: URLSessionTask!
+
+            switch self {
+            case let .request(urlRequest):
+                let urlRequest = urlRequest.adapt(using: adapter)
+                queue.sync { task = session.dataTask(with: urlRequest) }
```

Comment:
You'd be able to use it here like:
```swift
task = queue.syncAround { session.dataTask(with: urlRequest) }
```
Naming can be changed, but I saw the IUO -> sync -> return dance a few times in these changes.